### DISCUSSION TITLE
Typed variables in start_session and send_broadcast actions

### DISCUSSION THREAD

**Discussion by rowanseymour** [nyaruka/floweditor#1110] [Ideas]
Moved from https://github.com/nyaruka/goflow/issues/661

**Comment by ericnewcomer** [comment]
So is the thinking that groups and contacts would accept strings? Not sure how this doesn't also have a goflow component, is there already support for that?

**Comment by nicpottier** [comment]
For contacts I think it is URN (and maybe UUID?)
For groups maybe it is group name or UUID?

I think those are basically what is currently supported in `legacy_vars` but we want to type a bit more. (or we just maintain current behavior of trying in priority order)

**Comment by ericnewcomer** [comment]
Okay, so contacts can already take urns that are expressions? And groups can already accept names that are expressions? And both without uuids?

**Comment by rowanseymour** [comment]
A group reference can be "concrete", e.g.

```json
{
    "uuid": "",
    "name": "Testers"
}
```

or "variable", e.g.

```json
{
    "name_match": "@results.group"
}
```

which is how it works in add|remove group actions. Question is if we can use it that way in these other actions too, and what we'd do for matching contacts by tel and uuid.
